stats.go

```diff
@@ -126,12 +146,49 @@ func (t *timer) Record(interval time.Duration) {
 	t.reporter.ReportTimer(t.name, t.tags, interval)
 }
 
+func (t *timer) snapshot() []time.Duration {
+	t.unreported.RLock()
+	snap := make([]time.Duration, len(t.unreported.values))
+	for i := range t.unreported.values {
+		snap[i] = t.unreported.values[i]
+	}
+	t.unreported.RUnlock()
+	return snap
```

Comment:
Again, do we just want to have more and more timers building up between snapshots?